service/storage_service.cc

```diff
@@ -2658,20 +2658,207 @@ void storage_service::add_expire_time_if_found(inet_address endpoint, int64_t ex
+                    slogger.debug("load_and_stream: ops_uuid={}, ks={}, table={}, remove sst={}", ops_uuid, ks_name, cf_name, f);
+                    remove_file(f);
+                }
+            }
+            auto duration = std::chrono::duration_cast<std::chrono::duration<float>>(std::chrono::steady_clock::now() - start_time).count();
+            for (auto& [node, meta] : metas) {
+                slogger.info("load_and_stream: ops_uuid={}, ks={}, table={}, target_node={}, num_partitions_sent={}, num_bytes_sent={}",
+                        ops_uuid, ks_name, cf_name, node, meta.num_partitions_sent(), meta.num_bytes_sent());
+            }
+            auto partition_rate = std::fabs(duration) > FLT_EPSILON ? num_partitions_processed / duration : 0;
+            auto bytes_rate = std::fabs(duration) > FLT_EPSILON ? num_bytes_read / duration / 1024 / 1024 : 0;
+            slogger.info("load_and_stream: finished ops_uuid={}, ks={}, table={}, partitions_processed={} Partitions, bytes_processed={} Bytes, partitions_per_second={} Partitions/s, bytes_per_second={} MiB/s, duration={} s",
+                    ops_uuid, ks_name, cf_name, num_partitions_processed, num_bytes_read, partition_rate, bytes_rate, duration);
+        }
+    });
+}
+
+// For more details, see the commends on column_family::load_new_sstables
+// All the global operations are going to happen here, and just the reloading happens
+// in there.
+future<> storage_service::load_new_sstables(sstring ks_name, sstring cf_name,
+        bool load_and_stream, bool primary_replica_only) {
+    seastar::thread_attributes attr;
+    attr.sched_group = _db.local().get_streaming_scheduling_group();
+    return seastar::async(std::move(attr), [this, ks_name, cf_name, load_and_stream, primary_replica_only] {
+        if (_loading_new_sstables) {
+            throw std::runtime_error("Already loading SSTables. Try again later");
+        } else {
+            _loading_new_sstables = true;
+        }
+        slogger.info("Loading new SSTables for keyspace={}, table={}, load_and_stream={}, primary_replica_only={}",
+                ks_name, cf_name, load_and_stream, primary_replica_only);
+        try {
+            if (load_and_stream) {
+                utils::UUID table_id;
+                std::vector<std::vector<sstables::shared_sstable>> sstables_on_shards;
+                std::tie(table_id, sstables_on_shards) = distributed_loader::get_sstables_from_upload_dir(_db, ks_name, cf_name).get0();
+                service::get_storage_service().invoke_on_all([&sstables_on_shards, ks_name, cf_name, table_id, primary_replica_only] (storage_service& ss) mutable {
+                    return ss.load_and_stream(ks_name, cf_name, table_id, std::move(sstables_on_shards[this_shard_id()]), primary_replica_only);
+                }).get();
+            } else {
+                distributed_loader::process_upload_dir(_db, _sys_dist_ks, _view_update_generator, ks_name, cf_name).get();
```

Comment:
How is view update generation handled? By the current streaming server code?

Reply:
Like rebuild or replace,  the receiver does not generate view update, the view table should be loaded and streamed directly.

Reply:
But this isn't how it works with regular nodetool refresh.

I don't know if it's good or bad, if users are expected to keep backups of view tables or not.